jscomp/others/jsx.ml

```diff
@@ -38,8 +38,5 @@ type 'props component = ('props, element) componentLike
 external component : ('props, element) componentLike -> 'props component
   = "%identity"
 
-let addKeyProp (p : 'props) (k : string) : 'props =
-  (* Use Js_obj.assign, not Js.Obj.assign, otherwise the dependency will not be picked up correctly
-     in the ninja file. *)
-  Obj.magic (Js_obj.assign (Obj.magic p) [%obj { key = k }])
-  [@@inline]
+external addKeyProp : 'props -> string -> 'props = "Object.assign"
```

Comment:
There's nothing left to say this is "key".

Reply:
Perhaps one could use raw instead, and use directly to create functions such as `createElementVariadicWithKey`

CC @cknitt do you have a revised suggestion?

Reply:
Or even without raw, if you move `addKeyProp` to the same place where `createElementVariadicWithKey` is defined, it will be inlined.

Reply:
> Or even without raw, if you move `addKeyProp` to the same place where `createElementVariadicWithKey` is defined, it will be inlined.

Yes, it will be inlined if `addKeyProp` fn is located in rescript-react.

Reply:
How about just remove it from here.
Sorry for suggesting earlier to put it here.

Reply:
> How about just remove it from here.
> Sorry for suggesting earlier to put it here.

No problem! I’ll remove it here and put it in the same place of creatElement.